src/hevm/src/EVM/SymExec.hs

```diff
@@ -36,14 +37,16 @@ import qualified Data.Text.Lazy.IO as TL
 import EVM.Format (formatExpr)
 
 data ProofResult a b c = Qed a | Cex b | Timeout c
-  deriving (Show)
+  deriving (Show, Eq)
 type VerifyResult = ProofResult (Expr End) (Expr End, SMTCex) (Expr End)
-type EquivalenceResult = ProofResult ([VM], [VM]) VM ()
 
 isQed :: ProofResult a b c -> Bool
 isQed (Qed _) = True
 isQed _ = False
 
+containsA :: ProofResult () () () -> [(a , b, ProofResult () () ())] -> Bool
```

Comment:
```suggestion
containsA :: ProofResult a b c -> [(d, e, ProofResult a b c)] -> Bool
```

I think this can be made a bit more general?

Reply:
Good point, thanks, fixed!